test/rubygems/test_gem_gemcutter_utilities.rb

```diff
@@ -345,4 +348,75 @@ def test_verify_missing_api_key
       @cmd.verify_api_key :missing
     end
   end
+
+  class SignInFetcher < Gem::FakeFetcher
```

Comment:
`util_gem_signin` was getting too gnarly to add fetcher data so I extracted a helper class to setup the fetcher. After writing the rest of the tests for the other commands, I'm intending to follow up with another refactor to have a universal child fetcher for webauthn verifications since they all stub the same requests.

Reply:
Done in https://github.com/rubygems/rubygems/pull/6774/commits/f49fb73cac9b6700b7de07fd977a5c21578046cf